pkg/operator/starter.go

```diff
@@ -26,19 +35,96 @@ const (
 	instanceName     = "csi.ovirt.org"
 )
 
-func RunOperator(ctx context.Context, controllerConfig *controllercmd.ControllerContext) error {
+type CSIOperator struct {
+	ovirtClient  *ovirt.Client
+	nodeName     *string
+	storageClass *storagev1.StorageClass
+}
+
+func NewCSIOperator(nodeName *string) (*CSIOperator, error) {
+	client, err := ovirt.NewClient()
+	if err != nil {
+		klog.Error(err)
+		return nil, err
+	}
+
+	return &CSIOperator{
+		ovirtClient: client,
+		nodeName:    nodeName,
+	}, nil
+}
+
+func (o *CSIOperator) getStorageDomain(ctx context.Context, kubeClient *kubeclient.Clientset) string {
+	get, err := kubeClient.CoreV1().Nodes().Get(ctx, *o.nodeName, metav1.GetOptions{})
+	if err != nil {
+		klog.Fatal(err)
```

Comment:
I am not very comfortable about this. Errors happen. The code should be resilient and handle it. That's why I suggested that you should write your own controller. It's not much more code than it's already here, just structured differently.

Reply:
ACK, removed and rewritten